pkg/network/dns.go

```diff
@@ -1,11 +1,14 @@
 package network
 
-import "github.com/DataDog/datadog-agent/pkg/process/util"
+import (
+	"github.com/DataDog/datadog-agent/pkg/process/util"
+	"github.com/google/gopacket/layers"
+)
 
 // ReverseDNS translates IPs to names
 type ReverseDNS interface {
 	Resolve([]ConnectionStats) map[util.Address][]string
-	GetDNSStats() map[DNSKey]map[string]DNSStats
+	GetDNSStats() map[DNSKey]map[string]map[layers.DNSType]DNSStats
```

Comment:
Can we create our own type for query type? That way we aren't leaking the type of the parsing library, which is an implementation detail. This should be as easy as `type QueryType layers.DNSType`.

Reply:
we could; however, we'd also have to map in all the `const` values for the DNSType, which at least in theory could allow them to get out of sync.

Reply:
They are based on the DNS RFC, and you can directly map them so they don't get out of sync. My main concern is that any user of this function would also need to import `github.com/google/gopacket/layers`, which we might not use in the future if we start doing the DNS packet decoding in eBPF.